skate-plugin/src/main/kotlin/com/slack/sgp/intellij/ui/WhatsNewPanelFactory.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2023 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.sgp.intellij.ui
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.runReadAction
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.wm.ToolWindow
+import com.intellij.testFramework.LightVirtualFile
+import com.intellij.ui.content.ContentFactory
+import java.awt.BorderLayout
+import javax.swing.JComponent
+import javax.swing.JPanel
+import org.intellij.lang.annotations.Language
+import org.intellij.plugins.markdown.ui.preview.html.MarkdownUtil
+import org.intellij.plugins.markdown.ui.preview.jcef.MarkdownJCEFHtmlPanel
+
+/**
+ * The WhatsNewPanelFactory class takes the markdown file string from SkateService and displays it
+ * in a tool window. It uses MarkdownJCEFHtmlPanel and has dependency on intellij markdown plugin to
+ * properly format the markdown file and its contents
+ */
+class WhatsNewPanelFactory : DumbAware {
+
+  // Function that creates the tool window
+  fun createToolWindowContent(
+    toolWindow: ToolWindow,
+    project: Project,
+    markdownFileString: String,
+    parentDisposable: Disposable
+  ) {
+
+    val toolWindowContent = WhatsNewPanelContent(project, markdownFileString, parentDisposable)
+    val content =
+      ContentFactory.getInstance().createContent(toolWindowContent.contentPanel, "", false)
+    toolWindow.contentManager.addContent(content)
+  }
+
+  private class WhatsNewPanelContent(
+    project: Project,
+    @Language("Markdown") markdownFileString: String,
+    parentDisposable: Disposable
+  ) {
+
+    // Actual panel box for What's New at Slack
+    val contentPanel: JPanel =
+      JPanel().apply {
+        layout = BorderLayout(0, 20)
+        add(createControlsPanel(project, markdownFileString, parentDisposable), BorderLayout.CENTER)
+      }
+
+    // Control Panel that takes in the current project, markdown string, and a Disposable.
+    // The Disposable is necessary to prevent a substantial memory leak while working with
```

Comment:
We don't do anything with the parent disposable above though, could you elaborate on the leak it's preventing and how?

Reply:
I had actually gotten that information from here: https://intellij-support.jetbrains.com/hc/en-us/community/posts/10856308259090-Markdown-Preview-Swing-Component. This is also where I got how to implement MarkdownJCEFHtmlPanel.

Reply:
So this only prevents a leak if we dispose the panel somewhere, but I don't see that happening in this code. It looks like we only create the disposable. A disposable is a common concept used for cases where you have some process or resource and you want to be able to "dispose" of it later. We use it a lot in RxJava too. It's only doing its job if we call `dispose()` somewhere though, so I think we still have more to do here

Reply:
Should I be disposing it in `SkateService` after the tool window is closed maybe?